### PR TITLE
Add JetBrains Gateway setup details

### PR DESCRIPTION
**Pull request by kotwanikunal**
### Description
Adds setup details for remote IDE support using JetBrains Gateway
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
     - [IntelliJ IDEA](#intellij-idea)
     - [Visual Studio Code](#visual-studio-code)
     - [Eclipse](#eclipse)
+    - [JetBrains Gateway](#jetbrains-gateway)
```
Comment (owaiskazi19) [inline DEVELOPER_GUIDE.md:17]: Shouldn’t this be just under https://github.com/opensearch-project/OpenSearch/pull/1944/files#diff-d5555724c2ebca1b1f9be92a74c90e84c2dcbc76631148801024e24372b7edc6R14 and no separate index for the same?

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -202,6 +203,15 @@ You can now import the OpenSearch project into Eclipse as follows.
 
 **Note:** it may look non-intuitive why one needs to use Gradle wrapper and then import existing Gradle project (in general, **File > Import -> Existing Gradle Project** should be enough). Practically, as it stands now, Eclipse Buildship plugin does not import OpenSearch project dependencies correctly but does work in conjunction with Gradle wrapper.
 
+### JetBrains Gateway
+
+[JetBrains Gateway](https://www.jetbrains.com/remote-development/gateway/) enables development, testing and debugging on remote machines like development servers.
+
+1. On the local development machine, download and install the latest thin client from the [JetBrains Gateway page](https://www.jetbrains.com/remote-development/gateway/)
```
Comment (owaiskazi19) [inline DEVELOPER_GUIDE.md:210]: Missing period.

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -202,6 +203,15 @@ You can now import the OpenSearch project into Eclipse as follows.
 
 **Note:** it may look non-intuitive why one needs to use Gradle wrapper and then import existing Gradle project (in general, **File > Import -> Existing Gradle Project** should be enough). Practically, as it stands now, Eclipse Buildship plugin does not import OpenSearch project dependencies correctly but does work in conjunction with Gradle wrapper.
 
+### JetBrains Gateway
+
+[JetBrains Gateway](https://www.jetbrains.com/remote-development/gateway/) enables development, testing and debugging on remote machines like development servers.
```
Comment (saratvemulapalli) [inline DEVELOPER_GUIDE.md:208]: Just curious, this seems to be nothing specific to opensearch. Do we still want to have this in the docs?

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (kotwanikunal) [reply DEVELOPER_GUIDE.md:17]: Yes, it makes sense to add it under remote development for IntelliJ

File: DEVELOPER_GUIDE.md (hunk 3 above)
Comment (kotwanikunal) [reply DEVELOPER_GUIDE.md:208]: This is something which helped me as a contributor to get started quickly with remote debugging and development, given the nature of the server requirements for OpenSearch.

Comment (dblock) [reply DEVELOPER_GUIDE.md:208]: I think the confusing part is that JetBrains Gateway is not an editor. How about moving this section under "IntelliJ IDEA"?

VSCode has a similar remote host setup out of the box with no additional work, so we don't mention it.

Comment (kotwanikunal) [reply DEVELOPER_GUIDE.md]: Moved the section under remote development for IntelliJ.

**Review by dblock — APPROVED** [review]

**Pull request merged**